clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -916,12 +918,18 @@ public void handleCompletedMetadataResponse(RequestHeader requestHeader, long no
             this.metadataFetchInProgress = false;
             Cluster cluster = response.cluster();
 
-            long missingListenerCount = response.topicMetadata().stream()
-                    .flatMap(topicMetadata -> topicMetadata.partitionMetadata().stream())
+            // If any partition has leader with missing listeners, log a few for diagnosing broker configuration
+            // issues. This could be a transient issue if listeners were added dynamically to brokers.
```

Comment:
If it could be transient, maybe it should be `warn` instead of `error`?

Reply:
@ijuma Thanks for the review, updated.